tests/Unit/ControlSystem/Test_TimescaleTuner.cpp

```diff
@@ -0,0 +1,342 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ControlSystem/TimescaleTuner.hpp"
+
+#include <array>
+#include <cmath>
+
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "tests/Unit/TestingFramework.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.TimescaleTuner.Test",
+                  "[ControlSystem][Unit]") {
+  const double max_threshold = 1.0e-2;
+  const double min_threshold = 1.0e-4;
+  const double increase_factor = 1.01;
+  const double decrease_factor = 0.99;
+  const double max_timescale = 10.0;
+  const double min_timescale = 1.0e-3;
+
+  const DataVector init_timescale{1.0};
+  TimescaleTuner tst(init_timescale, max_timescale, min_timescale,
+                     max_threshold, min_threshold, increase_factor,
+                     decrease_factor);
+
+  const DataVector timescale{1.0};
+  CHECK(tst.current_timescale() == timescale);
+
+  // Check the suggested timescale function
+  // (1) timescale < min_timescale
+  double tscale = 1.0e-4;
```

Comment:
Rather than hard code 1.0e-4, might be clearer to set it to min_timescale / 10.0